crypto/aria/aria.c

```diff
@@ -1,454 +1,1097 @@
-        return -1;
-    memcpy(w0, userKey, sizeof(w0));
-    switch (bits) {
-    default:
-        return -2;
-    case 128:
-        key->rounds = 12;
-        ck1 = c1;
-        ck2 = c2;
-        ck3 = c3;
-        memset(kr, 0, sizeof(kr));
-        break;
-
-    case 192:
-        key->rounds = 14;
-        ck1 = c2;
-        ck2 = c3;
-        ck3 = c1;
-        memcpy(kr, userKey + ARIA_BLOCK_SIZE, sizeof(kr) / 2);
-        memset(kr + ARIA_BLOCK_SIZE / 2, 0, sizeof(kr) / 2);
-        break;
-
-    case 256:
-        key->rounds = 16;
-        ck1 = c3;
-        ck2 = c1;
-        ck3 = c2;
-        memcpy(kr, userKey + ARIA_BLOCK_SIZE, sizeof(kr));
-        break;
-    }
-
-    FO(w3, w0, ck1);	xor128(w1, w3, kr);
-    FE(w3, w1, ck2);	xor128(w2, w3, w0);
-    FO(kr, w2, ck3);	xor128(w3, kr, w1);
-
-    rot19r(key->rd_key[ 0], w0, w1);
-    rot19r(key->rd_key[ 1], w1, w2);
-    rot19r(key->rd_key[ 2], w2, w3);
-    rot19r(key->rd_key[ 3], w3, w0);
-
-    rot31r(key->rd_key[ 4], w0, w1);
-    rot31r(key->rd_key[ 5], w1, w2);
-    rot31r(key->rd_key[ 6], w2, w3);
-    rot31r(key->rd_key[ 7], w3, w0);
-
-    rot61l(key->rd_key[ 8], w0, w1);
-    rot61l(key->rd_key[ 9], w1, w2);
-    rot61l(key->rd_key[10], w2, w3);
-    rot61l(key->rd_key[11], w3, w0);
-
-    rot31l(key->rd_key[12], w0, w1);
-    if (key->rounds > 12) {
-        rot31l(key->rd_key[13], w1, w2);
-        rot31l(key->rd_key[14], w2, w3);
-
-        if (key->rounds > 14) {
-            rot31l(key->rd_key[15], w3, w0);
-            rot19l(key->rd_key[16], w0, w1);
-        }
-    }
-    return 0;
-}
-
-/*
- * Expand the cipher key into the decryption key schedule.
- */
-int aria_set_decrypt_key(const unsigned char *userKey, const int bits,
-                         ARIA_KEY *key)
-{
-    ARIA_KEY ek;
-    const int r = aria_set_encrypt_key(userKey, bits, &ek);
-    unsigned int i, rounds = ek.rounds;
-
-    if (r == 0) {
-        key->rounds = rounds;
-        memcpy(key->rd_key[0], ek.rd_key[rounds], sizeof(key->rd_key[0]));
-        for (i = 1; i < rounds; i++)
-            a(key->rd_key[i], ek.rd_key[rounds - i]);
-        memcpy(key->rd_key[rounds], ek.rd_key[0], sizeof(key->rd_key[rounds]));
-    }
-    return r;
-}
+/*
+ * Copyright 2017 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+/* 
+ * Copyright (C) 2017 National Security Research Institute. All Rights Reserved.
+ *
+ * Information for ARIA
+ *     http://210.104.33.10/ARIA/index-e.html (English)
+ *     http://seed.kisa.or.kr/ (Korean)
+ *
+ * Public domain version is distributed above.
+ */
+
+/* ====================================================================
+ * Copyright (c) 2017 Oracle and/or its affiliates.  All rights reserved.

```

Comment:
This code contains both the new 32-bit implementation and the previous 8-bit implementation via OPENSSL_SMALL_FOOTPRINT flag. The line mentioned is from the previous commit.